### PR TITLE
Metric unit fixes

### PR DESCRIPTION
**Pull request by skonto**
# Changes
- Fixes some deprecated units (UnitOne -> UnitDimensionless) to be aligned with other repos eg. Eventing and a unit set wrongly.

/cc @evankanderson @vaikas 
/kind Bug

### CODE REVIEW

File: controller/stats_reporter.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func init() {
 		Retries: stats.Int64(
 			"workqueue_retries_total",
 			"Total number of retries handled by workqueue",
-			"s",
+			stats.UnitDimensionless,
```
Comment (skonto) [inline controller/stats_reporter.go:76]:
```# HELP controller_workqueue_retries_total Total number of retries handled by workqueue
# TYPE controller_workqueue_retries_total counter
controller_workqueue_retries_total{name="knative.dev-serving-pkg-reconciler-labeler.Reconciler-fast"} 3
controller_workqueue_retries_total{name="knative.dev-serving-pkg-reconciler-revision.Reconciler-fast"} 1260
controller_workqueue_retries_total{name="knative.dev-serving-pkg-reconciler-serverlessservice.reconciler-fast"} 2```

**Review by antoineco — APPROVED** [review]
/lgtm

File: controller/stats_reporter.go [hunk 2]
```diff
@@ -99,49 +99,49 @@ func init() {
...
 			"reflector_short_watches_total",
 			"Total number of short API watches done by the reflectors",
-			stats.UnitNone,
+			stats.UnitDimensionless,
 		),
 		WatchDuration: stats.Float64(
 			"reflector_watch_duration_seconds",
 			"How long an API watch takes to return and decode for the reflectors",
-			stats.UnitNone,
+			stats.UnitDimensionless,
```
Comment (skonto) [inline controller/stats_reporter.go:139]: Probably need to fix this one too s/UnitDimensionless to "s"

Comment (antoineco) [reply controller/stats_reporter.go:139]: Or `UnitSeconds`

**Review by antoineco — APPROVED** [review]
/lgtm

File: controller/stats_reporter.go [hunk 3]
```diff
@@ -99,49 +99,49 @@ func init() {
...
 			"reflector_last_resource_version",
 			"Last resource version seen for the reflectors",
-			stats.UnitNone,
+			stats.UnitDimensionless,
 		),
 		ListDuration: stats.Float64(
 			"reflector_list_duration_seconds",
 			"How long an API list takes to return and decode for the reflectors",
-			stats.UnitNone,
+			stats.UnitSeconds,
```
Comment (skonto) [inline controller/stats_reporter.go:124]: To be aligned with the rest of the metrics this needs `seconds` as unit, same for `reflector_watch_duration_seconds`.

File: controller/stats_reporter.go (hunk 2 above)
Comment (skonto) [reply controller/stats_reporter.go]: Yeah I just updated all of "s" instances ;)

**Comment by skonto** [comment]
```
=== RUN   TestMetricsExport/Prometheus
    e2e_test.go:320: unable to init: timed out waiting for the condition
    --- FAIL: TestMetricsExport/Prometheus (10.06s)
```
We need to spend some cycles to fix this one.

**Review by vagababov — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**